src/supplemental/nanolib/hocon.c

```diff
@@ -66,7 +66,9 @@ path_expression_parse(cJSON *jso)
 			path_expression_parse(child);
 		}
 		if (NULL != child->string &&
-		    NULL != strchr(child->string, '.')) {
+		    NULL != strchr(child->string, '.') &&
+		    (parent->string != NULL &&
+		        strncmp(parent->string, "password", 8))) {
```

Comment:
this may cause potential issue when extend the usage of conf, especially when using `password` as parent .
Or we could just suggest users to avoid using dot in items.